File: Peststore-Api-Automation/tests/petTests.js
const chakram = require('chakram'), expect = chakram.expect;

describe("Pet Module", function() {

    it("GET - GetPetInfo", function () {
        const resBody = {
            "id": 30,
            "category": {
            "id": 50,
            "name": "reptiles"
            },
            "name": "gecko",
            "photoUrls": [
            "https://example.com/reptiles.jpg"
            ],
            "tags": [
            {
                "id": 3,
                "name": "cute"
            }
            ],
            "status": "available"
        };

        const response = chakram.get("https://petstore.swagger.io/v2/pet/30");
        expect(response).to.have.status(200);
        expect(response).to.comprise.of.json(resBody);
        expect(response).to.have.header('content-type', 'application/json'); 
        return chakram.wait();
    });  
   
    it("PUT - UpdatePet", function () {
        const resBody = {
            "code": 405,
            "type": "unknown"
        };

        const body = {
            "id": 20,
            "category": {
            "id": 4,
            "name": "Cats"
            },
            "name": "Scottish",
            "photoUrls": [
            "https://example.com/updated-photo.jpg"
            ],
            "tags": [
            {
            "id": 7,
            "name": "Updated Tag"
            }
            ],
            "status": "available"
        };

        const response = chakram.put("https://petstore.swagger.io/v2/pet/70", body);
        expect(response).to.have.status(405);
        expect(response).to.comprise.of.json(resBody); 
        expect(response).to.have.header('content-type', 'application/json'); 
        return chakram.wait();
    });
   
    it("DELETE - DeletePet", function () {
       const resBody = {
           "code": 200,
           "type": "unknown",
           "message": "30"
        };

        const response = chakram.delete("https://petstore.swagger.io/v2/pet/30");
        expect(response).to.have.status(200);
        expect(response).to.comprise.of.json(resBody); 
        expect(response).to.have.header('content-type', 'application/json'); 
        return chakram.wait();
    });
   
   it("GET - GetPetInfo Deleted Pet", function () {
       const resBody = {
           "code": 1,
           "type": "error",
           "message": "Pet not found"
        };

        const response = chakram.get("https://petstore.swagger.io/v2/pet/30");
        expect(response).to.have.status(404);
        expect(response).to.comprise.of.json(resBody); 
        expect(response).to.have.header('content-type', 'application/json'); 
        return chakram.wait();
    });

   it("POST - CreatePet", function () {
       const resBody = {
           "id": 30,
           "category": {
           "id": 50,
           "name": "reptiles"
           },
           "name": "gecko",
           "photoUrls": ["https://example.com/reptiles.jpg"  
           ],
           "tags": [
           {
            "id": 3,
            "name": "cute"
           }
           ],
           "status": "available"
        };

       const body = {
           "id": 30,
           "category": {
           "id": 50,
           "name": "reptiles"
           },
           "name": "gecko",
           "photoUrls": ["https://example.com/reptiles.jpg"  
           ],
           "tags": [
           {
            "id": 3,
            "name": "cute"
           }
           ],
           "status": "available"
        };

        const response = chakram.post("https://petstore.swagger.io/v2/pet", body);
        expect(response).to.have.status(200);
        expect(response).to.comprise.of.json(resBody); 
        expect(response).to.have.header('content-type', 'application/json'); 
        return chakram.wait();
    });
});

File: Petstore-Load-Test/petTests.py
from locust import HttpUser, between, task

class WebsitePet(HttpUser):
    wait_time = between(5, 15)

    def addPetFunction(self, id):
        payload = {
            "id": id,
            "category": {
                 "id": 55,
                 "name": "cats"
             },
            "name": "scottish",
            "photoUrls": [
                    "example"
            ],
            "tags": [
                    {
                     "id": 555,
                     "name": "tag_name"
                     }
            ],
            "status": "available"
        }
        response = self.client.post("/v2/pet", json=payload)
        return response

    @task
    def addPet(self):
        payload = {
            "id": 5,
            "category": {
                 "id": 55,
                 "name": "cats"
             },
            "name": "scottish",
            "photoUrls": [
                    "example"
            ],
            "tags": [
                    {
                     "id": 555,
                     "name": "tag_name"
                     }
            ],
            "status": "available"
        }
        self.client.post("/v2/pet", json=payload)

    @task
    def getPetInfo(self):
        self.addPetFunction(5)
        self.client.get("/v2/pet/5")

    @task
    def deletePetInfo(self):
        self.addPetFunction(9)
        self.client.delete("/v2/pet/9")

    @task
    def updatePetInfo(self):
        body = {
            "id": 7,
            "category": {
                "id": 55,
                "name": "cats"
            },
            "name": "scottish",
            "photoUrls": [
                "example"
            ],
            "tags": [
                {
                    "id": 555,
                    "name": "tag_name"
                }
            ],
            "status": "available"
        }
        self.client.put("/v2/pet", json=body)

